colorcoder/ColorPairHelper.java

```diff
@@ -0,0 +1,58 @@
+package colorcoder;
+
+public class ColorPairHelper {
+
+   private static ColorPair GetColorFromPairNumber(int pairNumber) {
+        int zeroBasedPairNumber = pairNumber - 1;
+        int majorColorIndex = zeroBasedPairNumber / ColorPair.numberOfMajorColors ;
+        int minorColorIndex = zeroBasedPairNumber % ColorPair.numberOfMinorColors ;
+        ColorPair colorPair = retrieveColorPairBasedOnColorIndices(majorColorIndex,minorColorIndex);
+        return colorPair;
+    }
+
+   private static int GetPairNumberFromColor(MajorColor major,
+                                             MinorColor minor) {
+        return major.getIndex() * ColorPair.numberOfMinorColors + minor.getIndex() + 1;
+    }
+
+    public static void testNumberToPair(int pairNumber,
+                                 MajorColor expectedMajor,
+                                 MinorColor expectedMinor)
+    {
+        ColorPair colorPair = GetColorFromPairNumber(pairNumber);
+        System.out.println("Got pair " + colorPair);
+        assert(colorPair.getMajor() == expectedMajor);
+        assert(colorPair.getMinor() == expectedMinor);
+    }
+
+    public static void testPairToNumber(
```

Comment:
Test code is made as part of production code. The test code should be segregated to a separate file which hence can be excluded from deployment code. This class needs a re design to keep production code and test code separate.

Reply:
Hi Pritham. I have a query here I got your point about Test Code,  
1. I didn't see any junit dependency along with project code to write down test cases and keep the test code separate.
2. If Junit is not allowed to use, in that case do I need to create instance of Custom Test Class and invoke the custom test methods and any return type is expected from test methods regarding success or failure